scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,17 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                if self.keep_encoding_header:
+                    if isinstance(kwargs['flags'], list):
```

Comment:
`kwargs` is declared in line 46 above. `kwargs['flags']` will never be defined at this point, so this should be throwing a `KeyError` exception.